### PR TITLE
Phase2-hgx177 Two new methods to carry out trigger imulation for scintillators

### PR DESCRIPTION
**Pull request by bsunanda**
2 new methods: int getPhiBins(int lay) and std::pair<int,int> getREtaRange(int lay) are introduced to provide # of phi bins and minimum/maximum r/eta index for a given layer

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
+upgrade

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc [hunk 1]
```diff
@@ -411,6 +411,27 @@ std::vector<HGCalParameters::hgtrap> HGCalDDDConstants::getModules() const {
...
+    nphi      = hgpar_->nPhiBinBH_[type];
+  }
+  return nphi;
+}
+
+std::pair<int,int> HGCalDDDConstants::getREtaRange(int lay) const {
+  int irmin(0), irmax(0);
+  if (mode_ == HGCalGeometryMode::Trapezoid) {
+    int indx = layerIndex(lay,false);
+    if ((indx >=0) && (indx < (int)(hgpar_->iradMinBH_.size()))) {
```
Comment (cvuosalo) [inline Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]:
```suggestion
    if ((indx >=0) && (indx < static_cast<int>(hgpar_->iradMinBH_.size()))) {
```

Comment (cvuosalo) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: Our CMS coding rules recommend using C++ casting: [Rule 4.15](https://github.com/cms-sw/cms-sw.github.io/blob/a27560c6ecb214893b5ef31fc6dd935fb7c2288c/cms_coding_rules.md#4----technical-coding-rules-1)

**Comment by cvuosalo** [comment]
Are the differences in the comparison test a known issue? I don't see how this PR could create differences.

File: Geometry/HGCalCommonData/src/HGCalDDDConstants.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HGCalCommonData/src/HGCalDDDConstants.cc]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by kpedro88** [comment]
comparison differences are unrelated to this PR

**Comment by kpedro88** [comment]
+upgrade

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**